newrelic/newrelic.go

```diff
@@ -91,12 +93,17 @@ func ConfigAdminAPIKey(adminAPIKey string) ConfigOption {
 // ConfigRegion sets the New Relic Region this client will use.
 func ConfigRegion(r region.Name) ConfigOption {
 	return func(cfg *config.Config) error {
-		if region, ok := region.Regions[r]; ok {
-			regCopy := *region
-			return cfg.SetRegion(&regCopy)
+		reg, err := region.Get(r)
+		if _, ok := err.(*region.UnknownUsingDefaultError); ok {
+			// Log
+			log.Error(err.Error())
```

Comment:
we do not log anywhere else, so not sure this feels right... Downstream could do this instead

Reply:
This seems okay for now.  I suspect that we could return something of a composite error too if we wanted rather than logging here.  Like join  the two error strings with a `, ` or something.

Reply:
Error wrapping is a thing, but I generally believe that the error returned could be passed all the way back and be understood by the end-user of the Client

Reply:
Yeah, for sure agree about getting that error back to the caller.  I can't tell what it would break here, but we could set the default region and return the error about using the default region to the caller.  You could just have a `SetRegion()` call right above this log message, and instead of log, return the error.

Reply:
I removed this.  We fail fast everywhere else, and I think that is the right thing to do for the Client.  We do hand back an error.  If we want to add resiliency to the client + logging later we can do so in more detail.